src/attached_probe.cpp

```diff
@@ -519,8 +503,19 @@ bool AttachedProbe::resolve_offset_uprobe(bool safe_mode, bool has_multiple_aps)
   if (func_offset == 0)
     return true;
 
-  check_alignment(
-      probe_.path, symbol, sym_offset, func_offset, safe_mode, probe_.type);
+  std::string alignment_err;
+  bool alignment_valid = is_alignment_valid(
+      probe_.path, symbol, sym_offset, func_offset, probe_.type, alignment_err);
+
+  if (!alignment_valid) {
+    if (missing_probes == ConfigMissingProbes::error) {
+      throw FatalUserException(alignment_err);
+    } else if (missing_probes == ConfigMissingProbes::warn) {
+      LOG(WARNING) << alignment_err << ". Skipping probe.";
+    }
+    return false;
```

Comment:
Using `switch` here would probably be more readable and we'd get some compiler checks.